v1/message_test.go

```diff
@@ -49,6 +51,41 @@ func TestEncodeMessage(t *testing.T) {
 	require.NoError(t, err)
 	require.EqualValues(t, StatusMessage{
 		Message: testMessageStruct,
+		ID:      MessageID(data),
+	}, val)
+}
+
+func TestWrappedMessageWithSignature(t *testing.T) {
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	data, err := EncodeMessage(testMessageStruct)
+	require.NoError(t, err)
+	wrappedMessage, err := WrapMessageV1(data, key)
+	require.NoError(t, err)
+	// Decode it back to a struct because, for example, map encoding is non-deterministic
+	// and it is not possible to compare bytes.
+	val, err := DecodeMessage(wrappedMessage)
+	require.NoError(t, err)
+	require.EqualValues(t, StatusMessage{
+		Message:   testMessageStruct,
+		ID:        crypto.Keccak256(data),
```

Comment:
Don't we want to use `MessageID`? In case the implementation of it changes, we would need to fix all tests.